sherpa/astro/fake.py

```diff
@@ -21,111 +21,158 @@
+    bkg_models="model" arguments), and also include the response
+    necessary to convert to channels and counts (a combination of the
+    is_source and pileup_model arguments). Setting the bkg_models
+    argument to a PHA dataset is now handled by adding the background
+    to the data first, and then setting include_bkg_data to True.
 
     Examples
     --------
-        Estimate the signal from a 5000 second observation using the
-        ARF and RMF from "src.arf" and "src.rmf" respectively:
-
-        >>> set_source(1, xsphabs.gal * xsapec.clus)
-        >>> gal.nh = 0.12
-        >>> clus.kt, clus.abundanc = 4.5, 0.3
-        >>> clus.redshift = 0.187
-        >>> clus.norm = 1.2e-3
-        >>> fake_pha(1, 'src.arf', 'src.rmf', 5000)
-
-        Simulate a 1 mega second observation for the data and model
-        from the default data set. The simulated data will include an
-        estimated background component based on scaling the existing
-        background observations for the source. The simulated data
-        set, which has the same grouping as the default set, for
-        easier comparison, is created with the 'sim' label and then
-        written out to the file 'sim.pi':
-
-        >>> arf = get_arf()
-        >>> rmf = get_rmf()
-        >>> bkg = get_bkg()
-        >>> bscal = get_backscal()
-        >>> grp = get_grouping()
-        >>> qual = get_quality()
-        >>> texp = 1e6
-        >>> set_source('sim', get_source())
-        >>> fake_pha('sim', arf, rmf, texp, backscal=bscal, bkg=bkg,
-        ...          grouping=grp, quality=qual, grouped=True)
-        >>> save_pha('sim', 'sim.pi')
+
+    Simulate the data from an absorbed APEC model, where the response
+    is manually created (a "perfect" RMF and the ARF is flat, with a
+    break at 3.2 keV), and then use the Sherpa plot objects to display
+    the simulated data and model. Note that the exposure time must be
+    set (if the model normalization is set).
+
+    >>> import numpy as np
+    >>> from sherpa.astro.data import DataPHA
+    >>> from sherpa.astro.instrument import Response1D, create_arf, create_delta_rmf
+    >>> from sherpa.astro.xspec import XSphabs, XSapec
+    >>> gal = XSphabs()
+    >>> clus = XSapec()
+    >>> gal.nh = 0.12
+    >>> clus.kt = 4.5
+    >>> clus.abundanc = 0.3
+    >>> clus.redshift = 0.23
+    >>> clus.norm = 6.3e-4
+    >>> chans = np.arange(1, 101, dtype=np.int16)
+    >>> egrid = np.linspace(1, 5, 101)
+    >>> elo, ehi = egrid[:-1], egrid[1:]
+    >>> yarf = 100 * (elo < 3.2) + 50 * (elo >= 3.2)
+    >>> arf = create_arf(elo, ehi, yarf)
+    >>> rmf = create_delta_rmf(elo, ehi, e_min=elo, e_max=ehi)
+    >>> pha = DataPHA("faked", chans, chans * 0)
+    >>> pha.set_response(arf=arf, rmf=rmf)
+    >>> pha.exposure = 50000
+    >>> resp = Response1D(pha)
+    >>> full_model = resp(gal * clus)
+    >>> rng = numpy.random.default_rng()
+    >>> fake_pha(pha, full_model, rng=rng)
```

Comment:
I don't see an `rng` argument to this function. Something missing fro ma rebase/merge with #1735?

Reply:
Yeah - so this was where I started what became #1735 but when I separated the two I obviously missed something. We may want to add this example back once #1735 is merged (so it depends on which gets done first ;-)

Reply:
I've removed this for now.